### ISSUE TITLE
Implement use of validation data

### ISSUE THREAD

**Issue by jaclark5** [openforcefield/back-to-school-jen#52]
In Finlay's repo the [training and validation interchanges are processed together](https://github.com/fjclark/descent-workflow/blob/2f10332e52ea7f75cbca53c2fe83da6be5d27256/workflow/parameterise.py#L177).

Thus, both train/test Topologies are passed together to the loss function with only one of those datasets. The loss is calculated only for the dataset that is passed (a step is only taken if training data is used).

[The loss for validation set is computed and saved](https://github.com/fjclark/descent-workflow/blob/2f10332e52ea7f75cbca53c2fe83da6be5d27256/workflow/train.py#L351) but is not involved in the fitting process.

Should we:
1. Do the same thing
2. Process the interchanges for them independently
